seaworthy/testtools.py

```diff
@@ -0,0 +1,58 @@
+"""
+Some (optional) utilities for use with testtools.
+
+While seaworthy doesn't require testtools, we find it useful in downstream
+container tests we write with seaworthy. This module contains various bits and
+pieces to make seaworthy work better with testtools.
+"""
+
+from testtools.matchers import MatchesSetwise, MatchesStructure, Mismatch
+
+
+class PsTreeMismatch(Mismatch):
+    def __init__(self, row_fields, child_count, fields_mm, children_mm):
+        self.row_fields = row_fields
+        self.child_count = child_count
+        self.fields_mm = fields_mm
+        self.children_mm = children_mm
+
+    def describe(self):
+        rfs = ["%s=%r" % (k, v) for k, v in sorted(self.row_fields.items())]
```

Comment:
I think you've copied in a slightly older revision of this code. I tweaked some of the style of this stuff before we merged to `django-bootstrap`.